internal/docstore/doc.go

```diff
@@ -48,4 +48,44 @@
 // If a struct doesn't have a DocstoreRevision field, then the logic described above
 // won't apply to documents read and written with that struct. All writes with the
 // struct will succeed even if the document was changed since the last Get.
+//
+//
+// Queries
+//
+// Docstore supports querying within a collection. Call the Query method on
+// Collection to obtain a Query value, Then build your query by calling Query methods
+// like Where, Limit and so on. Finally, call the Get method on the query to execute it.
+// The result in an iterator, whose use is described below.
+//
+//     iter := coll.Query().Where("size", ">", 10).Limit(5).Get(ctx)
+//
+// The Where methods defines a filter condition, much like a WHERE clause in SQL. Conditions
+// are of the form "field op value", where field is any document field path (including dot-separated
+// paths), op is one of "=", ">", "<", ">=" or "<=", and value can be a string or number.
+//
+// Although you can make multiple Where calls, not all provider systems can handle them.
```

Comment:
"not all provider systems can handle them" sounds a bit too strong to me, it implies it won't work.

How about something like:

"You can make multiple Where calls. In some cases, parts of a Where clause may be processed on the client rather than natively by the provider, which may have performance implications for large result sets. See the provider-specific package doc for details."

Reply:
Done.